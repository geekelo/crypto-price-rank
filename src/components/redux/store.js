import { configureStore } from '@reduxjs/toolkit';
import themeSlice from './theme/themeSlice';
import quotesSlice from './quotes/quotesSlice';
import cryptosSlice from './crypto/cryptosSlice';

const store = configureStore({
  reducer: {
    theme: themeSlice,
    quotes: quotesSlice,
    cryptos: cryptosSlice,
  },
});

export default store;
