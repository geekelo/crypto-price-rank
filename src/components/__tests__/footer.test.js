import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux'; // Import MemoryRouter if needed
import store from '../redux/store';
import Footer from '../footer';

test('Home component matches snapshot', () => {
  const { asFragment } = render(
    <Provider store={store}>
      <Footer />
    </Provider>,
  );

  // Compare the rendered component to a stored snapshot
  expect(asFragment()).toMatchSnapshot();
});
