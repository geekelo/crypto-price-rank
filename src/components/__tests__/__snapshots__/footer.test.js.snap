// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="footer lightfooter"
  >
    <img
      alt="logo"
      src="logoicon.png"
      width="55"
    />
    <div
      class="icon lighticon"
    >
      <p
        class="tagtitle"
      >
        CRYPTO PRICE RANK
      </p>
      <p
        class="copyright"
      >
         Â© COPYRIGHT GEEKELO 2023
      </p>
    </div>
  </div>
</DocumentFragment>
`;
